File: app/src/main/java/com/kaigarrott/stopwatchkt/TaskExecutors.java
package com.kaigarrott.stopwatchkt;

import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

public class TaskExecutors {

    private static final Object LOCK = new Object();
    private static TaskExecutors sInstance;
    private final Executor db;

    private TaskExecutors(Executor db) {
        this.db = db;
    }

    public static TaskExecutors getInstance() {
        if(sInstance == null) {
            synchronized (LOCK) {
                sInstance = new TaskExecutors(Executors.newSingleThreadExecutor());
            }
        }
        return sInstance;
    }

    public Executor db() {
        return db;
    }
}

File: app/src/main/java/com/kaigarrott/stopwatchkt/TaskExecutors.kt
package com.kaigarrott.stopwatchkt

import java.util.concurrent.Executor
import java.util.concurrent.Executors

class TaskExecutors private constructor(val db: Executor) {

    companion object {

        val instance = TaskExecutors(Executors.newSingleThreadExecutor())

    }
}
